### DISCUSSION TITLE
should internal network clients use outgoing buffer or TCP/UDP client

### DISCUSSION THREAD

**Discussion by zarlo** [CosmosOS/Cosmos#2768] [General]
right now internal clients like DHCP and DNS use use the outgoing buffer

im look for feed back on if we should keep it like this or make them use TCP/UDP client

**Comment by EtherealTheDev** [comment]
y not both

**Reply by zarlo** [reply]
that would be a bit pointless

**Comment by AsertCreator** [comment]
whats the difference

**Reply by zarlo** [reply]
TCP/UDP client adds over head

**Reply by AsertCreator** [reply]
then no

**Comment by homecat1** [comment]
DHCP and DNS are both application level programs (5 levels of network).
This means they should not see or be aware of any kernel data-structure, but use tcp/udp instead.
